zoobc/zoobc-core
# fix logic used to validate spine blocks

Blocksmiths in Spine blocks are taken from ActiveRegistry array (cached), whish is filled using records from node_registry, but when a node starts dwloading spine blocks is not aware of the node_registry state (which is part of main block / transaction 'system').
To compute the valid blocksmiths for spine blocks we must use node_public_key table instead, that keeps track of changes in node registry between spine blocks (every node that is added or removed has a record in spine public key table that reference next spine block to be created).

## Description
A brief description of the pull request.

## Breakdown
- [x] Add strategy implementation for processing spine blocks (blocksmithStrategy)
- [x] Add initial logic for validating spine blocks using spine_public_key table instead of (cached) node_registry

## Reference Issue
Close #1444 

## Step to Test (optional)
- Run a new node on an existing network that has at least one change in registry (added/deleted node)
- Verify that the node finish dowloading all spine blocks
